tina/schema.tsx

```diff
@@ -163,6 +163,36 @@ export const schema = defineSchema({
                 },
               ],
             },
+            {
+              name: 'GraphQLCodeBlock',
+              label: 'GraphQL Code Block',
+              fields: [
+                {
+                  type: 'string',
+                  name: 'query',
+                  label: 'Query',
+                  description:
+                    'Paste GraphQL query here. # are auto-inserted as spcing placeholders.',
+                  ui: {
+                    component: 'textarea',
+                    format: (val?: string) => val && val.replaceAll('#', ' '),
+                    parse: (val?: string) => val && val.replaceAll(' ', '#'),
```

Comment:
this is a little bit hacky

Reply:
This is a workaround (placeholder string replacement) so that the content in the embed has 'rich-text' characteristics (usually not possible). 

In this case I needed spaces at the beginning of the line to be preserved. I couldn't get this working with string field types alone, and rich-text can't be nested in rich-text (current limitation of Tina).

If this is acceptable for now – I could mark it in the code as a temporary workaround until Tina rich-text nesting is possible (afaik it's currently being developed) and create an associated PBI?

Also open to other ideas.

Reply:
Workarounds are ok, just make sure to leave a comment with a link to a PBI where you have logged it.

https://www.ssw.com.au/rules/todo-tasks/